src/position.cpp

```diff
@@ -902,7 +897,7 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
   // Add gating piece
   if (is_gating(m))
   {
-      assert(gating_type(m) == HAWK || gating_type(m) == ELEPHANT);
+      assert(gating_type(m) >= HAWK && gating_type(m) <= QUEEN);
```

Comment:
As you mentioned enabling gating of queens will only require a few more changes, but since we do not add it yet in this PR, we should keep `gating_type(m) <= ELEPHANT` for consistency.